### ISSUE TITLE
fix(modelengine): track imported decode run loops during shutdown

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12620]
# fix(modelengine): make CloseAndWait track a scheduler started by AdmitImported

```routing
lane: modelengine
paths: ["internal/modelengine/native_pd.go", "internal/modelengine/native_pd_test.go"]
expected_steps: 4
```

## Current state

At public source SHA `64cf527c1ca925822357ec113c6af37c2108084c`,
`NativeScheduler.AdmitImported` starts the scheduler goroutine through `s.started.Do`
but does not set `s.runStarted`. `NativeScheduler.CloseAndWait` reads that flag and
returns nil immediately when it is false. A scheduler used only by native P/D imports
therefore reports that shutdown is complete while an imported lane is still executing.

`NativePDCluster.Admit` reaches this path after moving prompt KV to a decode worker.
The cluster's current `Close` calls `NativeScheduler.Close` without waiting, so this
specific false wait result affects hosts that use `DecodeWorker(...).Scheduler()` or
direct `AdmitImported` plus `CloseAndWait`. The native P/D path itself is real; this
audit found gateway metrics/test attachment but no default `fak serve` construction
of a `NativePDCluster`.

Priority: P2. Work class: `class:dev`. A deterministic synthetic-model witness
blocks one imported lane in model execution and gives `CloseAndWait` an already
canceled wait context. A correctly tracked running loop must return
`context.Canceled`; current source incorrectly returns nil. No hardware was used.

Duplicate searches across open and closed `anthony-chaudhary/fak` issues for
`AdmitImported CloseAndWait runStarted`, `native PD scheduler CloseAndWait returns
early`, and `imported sequence shutdown wait` found no matching issue. This is
distinct from #12617: that issue permits a new ordinary lane to enqueue after the
run loop has stopped; here the imported lane was admitted and executing before
shutdown, but the wait path does not know its run loop ever started. It is also
distinct from #12589's buffered-token receive in `ContinuousBatcher.Close`.

## Core through-line

Every path that starts `NativeScheduler.run` must publish the same run-start state,
so `CloseAndWait` waits for imported P/D lanes exactly as it waits for ordinary lanes.

## Gold-plating boundary

Limit the change to shared scheduler-start bookkeeping and a native P/D regression.
Do not redesign P/D routing, KV transfer, scheduler shutdown, prefill, decoding, or
cluster ownership. A tiny shared start helper is acceptable if it removes the two
start sequences without widening the package API.

## Done condition

- [ ] `AdmitImported` records that the scheduler run loop started.
- [ ] `CloseAndWait` with a canceled wait context returns `context.Canceled` while
      an imported lane is still blocked in model execution.
- [ ] After model execution is released, the canceled lane retires and a later
      `CloseAndWait` returns nil.
- [ ] Ordinary `Admit` start and shutdown behavior remains unchanged.

## Definition of Done

Done when ordinary and imported admission use consistent run-start bookkeeping,
the deterministic regression passes three times, and the package suite remains green.

## Witness

From the public repository root, with the audit test placed at
`internal/modelengine/audit_native_imported_close_wait_test.go`:

```text
go test ./internal/modelengine -run '^TestAuditNativeSchedulerCloseAndWaitTracksImportedRunLoop$' -count=3 -v -timeout=30s
```

Observed on Windows amd64 with Go 1.26.7:

```text
=== RUN   TestAuditNativeSchedulerCloseAndWaitTracksImportedRunLoop
    audit_native_imported_close_wait_test.go:51: CloseAndWait error = <nil>, want context.Canceled while imported lane is executing
--- FAIL: TestAuditNativeSchedulerCloseAndWaitTracksImportedRunLoop (0.00s)
```

The same failure occurred on all three repetitions. The canceled wait context is
the deterministic oracle: the model execution hook holds the run loop open, so
`CloseAndWait` can return nil only by taking its false `runStarted` early return.

## Verifiable Witness

```go
package modelengine

import (
	"context"
	"errors"
	"testing"

	"github.com/anthony-chaudhary/fak/internal/model"
)

func TestAuditNativeSchedulerCloseAndWaitTracksImportedRunLoop(t *testing.T) {
	m := model.NewSynthetic(SyntheticConfig())
	s := NewNativeScheduler(m)
	executeEntered := make(chan struct{})
	releaseExecute := make(chan struct{})
	released := false
	release := func() {
		if !released {
			released = true
			close(releaseExecute)
		}
	}
	defer release()
	s.beforeModelExecute = func(_ nativeSchedulerEventKind, _ *schedLane) {
		close(executeEntered)
		<-releaseExecute
	}

	sess := m.NewSession()
	prompt := []int{1, 2, 3}
	logits := sess.Prefill(prompt)
	req, err := s.AdmitImported(context.Background(), nil, ImportedSequence{
		Session: sess, Logits: logits, Prompt: prompt, Tool: "audit-imported-close",
	})
	if err != nil {
		t.Fatalf("AdmitImported: %v", err)
	}

	select {
	case <-executeEntered:
	}

	waitCtx, cancelWait := context.WithCancel(context.Background())
	cancelWait()
	err = s.CloseAndWait(waitCtx)
	if !errors.Is(err, context.Canceled) {
		release()
		for range req.Tokens() {
		}
		_, _ = req.Result()
		t.Fatalf("CloseAndWait error = %v, want context.Canceled while imported lane is executing", err)
	}

	release()
	for range req.Tokens() {
	}
	if _, err := req.Result(); !errors.Is(err, context.Canceled) {
		t.Fatalf("Result error = %v, want context.Canceled", err)
	}
	if err := s.CloseAndWait(context.Background()); err != nil {
		t.Fatalf("CloseAndWait after execution release: %v", err)
	}
}
```

After adding the regression in an isolated working context, commit it with the fix
once green, then run:

```text
go test ./internal/modelengine -run '^TestAuditNativeSchedulerCloseAndWaitTracksImportedRunLoop$' -count=3 -v -timeout=30s
go test ./internal/modelengine -count=1 -timeout=120s
```

## Likely files

- `internal/modelengine/native_pd.go`
- `internal/modelengine/native_pd_test.go`

Exact seams at the audited SHA:

- `internal/modelengine/native_pd.go:59-107` (`AdmitImported`), especially its
  scheduler start at line 105.
- `internal/modelengine/nativesched.go:434-440` (ordinary admission's correct
  `runStarted` publication, used as the parity reference).
- `internal/modelengine/nativesched.go:493-513` (`CloseAndWait`).
- `internal/modelengine/native_pd_test.go` (add the regression above).

## Suggested implementation steps

1. Add the isolated regression and retain it with the fix once green.
2. Make `AdmitImported` publish `runStarted` under `s.mu` before launching `run`,
   matching ordinary admission; optionally centralize the identical start sequence.
3. Verify shutdown while imported model execution is blocked and again after it retires.
4. Run the targeted witness and the `internal/modelengine` package suite.

## Lane

modelengine

## Expected steps

4

## Working spine

Imported P/D admission -> scheduler run starts -> run-start state published ->
CloseAndWait observes and waits for retirement.

## Why this is next

The P/D admission path silently bypasses the bookkeeping that makes the scheduler's
blocking shutdown API truthful.

## Parent context

Standalone follow-up finding from the 2026-09-09 core inference correctness audit.

## Acceptance gate

The three-count targeted regression and the `internal/modelengine` package suite pass.

## Closure binding

The resolving commit references the public issue and carries `(fak modelengine)`.

## Problem frame

- Centrality: Core
- P1 Context: preserved - deterministic false shutdown-completion result on the
  native P/D scheduler path; no duplicate found.
- P2 Net value: advanced - lets hosts wait reliably before releasing resources
  associated with imported decode work.
- P3 Adaptation: preserved - no token, routing, transfer, or sampling behavior changes.
- P4 Operations: preserved - one package, two likely files, four steps.

## Blast radius and fallback

Affected: `NativeScheduler` instances first started by `AdmitImported` and later
closed through `CloseAndWait`. Unaffected: ordinary `Admit`, which already sets the
flag; `NativePDCluster.Close`, which currently uses non-waiting `Close`; external
adapters; and model math. No production code or quarantine was added during audit.

**Comment by anthony-chaudhary** [comment]
Resolved in commit 33b02ebb0 (public fak main). Verification witness: `go test ./internal/modelengine -run '^TestNativeSchedulerCloseAndWaitTracksImportedRunLoop$' -count=3 -timeout=30s` -> PASS 3/3; `go test ./internal/modelengine -count=1 -timeout=300s` -> ok; `go vet ./internal/modelengine` -> clean; gofmt clean; fak-boundary check --staged -> PASS 0 violations. [SW-VERIFIED]: AdmitImported now publishes runStarted under s.mu before launching run, matching ordinary Admit; CloseAndWait with a canceled wait context returns context.Canceled while an imported lane is executing. No hardware used.
